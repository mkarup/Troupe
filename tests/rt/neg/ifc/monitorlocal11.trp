let val secret = false raisedTo `{secret}`
    val p = spawn (fn () =>
                      if secret then
                        receive [ hn _ => () ]
                      else
                        receive [ hn _ => 1 ^ "crash" ]
                  )
    val reason = 
        let pini authority
            val monRef = monitorlocal (p, `{secret}`)
            val _ = send (p, ())
            val reason = rcvp (`{secret}`, [ hn ("DONE", _, _, reason) => reason ])
        in
          reason
        end
in
  ladv(`{}`, reason)
end
  