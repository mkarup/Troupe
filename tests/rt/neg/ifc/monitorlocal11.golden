2024-12-11T16:27:06.847Z [RTM] [32minfo[39m: Skipping network creation. Observe that all external IO operations will yield a runtime error.
Runtime error in thread f105cb1c-db76-4f0e-b6a1-e31569aa1ddb@{}%{}
>> value 1 is not a string
Runtime error in thread ab7e4388-401d-44c8-a5b3-b84c2733f606@{}%{}
>> Illegal flow in adv function:
 |    pc: {}
 | block: {}
 | l_adv: {}@{}%{} 
 | value: (1@{secret}%{secret}, "value 1 is not a string"@{secret}%{secret})@{secret}%{secret}
